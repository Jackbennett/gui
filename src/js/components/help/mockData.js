export const helpProps = {
  docsVersion: '',
  findLocalIpAddress: jest.fn,
  isHosted: false,
  isEnterprise: false,
  menderDebPackageVersion: 'master',
  menderVersion: 'master',
  menderArtifactVersion: 'master',
  org: {
    tenant_token: ''
  }
};
