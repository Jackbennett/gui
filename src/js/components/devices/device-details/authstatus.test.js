import React from 'react';
import { render } from '@testing-library/react';
import AuthStatus from './authstatus';
import { defaultState, undefineds } from '../../../../../tests/mockData';

describe('AuthStatus Component', () => {
  it('renders correctly', async () => {
    const { baseElement } = render(
      <AuthStatus
        device={{
          ...defaultState.devices.byId.a1,

          auth_sets: [
            ...defaultState.devices.byId.a1.auth_sets,
            {
              ...defaultState.devices.byId.a1.auth_sets[0],
              status: 'pending'
            }
          ]
        }}
        toggleAuthsets={jest.fn}
      />
    );
    const view = baseElement.firstChild.firstChild;
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });
});
