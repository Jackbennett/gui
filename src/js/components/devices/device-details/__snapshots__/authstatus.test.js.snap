// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthStatus Component renders correctly 1`] = `
<div
  class="margin-bottom "
>
  <div
    class="clickable flexbox space-between"
    style="align-items: flex-start;"
  >
    <div
      class="flexbox center-aligned"
    >
      <h4>
        Authentication status
      </h4>
      <div
        class="flexbox center-aligned margin-left margin-right"
      >
        <div
          class="capitalized"
        >
          accepted
        </div>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root green"
          focusable="false"
          style="margin: 12px;"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
          />
        </svg>
      </div>
      <div
        class="MuiChip-root MuiChip-colorPrimary MuiChip-sizeSmall"
      >
        <span
          class="MuiChip-label MuiChip-labelSmall"
        >
          new request
        </span>
      </div>
    </div>
    <div
      class="flexbox centered"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  <div>
    <a>
      show more
    </a>
  </div>
  <hr
    class="MuiDivider-root"
    style="margin-top: 24px;"
  />
</div>
`;
